extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/JdbcBasePersistenceImpl.java

```diff
@@ -189,6 +185,17 @@ private void persistEntity(
     }
   }
 
+  private int execute(Object executor, String query) throws SQLException {
+    if (executor instanceof Statement) {
+      // used for running in transaction
+      return ((Statement) executor).executeUpdate(query);
+    } else if (executor instanceof DatasourceOperations) {
+      return ((DatasourceOperations) executor).executeUpdate(query);
+    } else {
+      throw new IllegalArgumentException("Unsupported executor: " + executor);
+    }
+  }
```

Comment:
Unrelated to the transaction discussions, this code pattern does not look nice from the object-oriented design POV. Would it be possible to avoid if/else with `instanceof` if you still want to achieve different call paths for these cases?

Reply:
i see, i tried the following :  try out passing a Consumer and sneakyThrow but it does introduce another set of if-elses so settled on this for now. Open to other suggestions. OOP is not possible honestly as these are not our managed object to define an intereface on and then calling interface function, but i think it was too much for the goal to achieve. Please let me know if you strongly feel about this, i can introduce that introduce interface as well